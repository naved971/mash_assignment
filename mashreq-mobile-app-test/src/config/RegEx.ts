export const RegEx : any = {
    USERNAME:/^[a-zA-Z0-9]{10,}$/,
    USERNAME_UAE: /^[a-zA-Z0-9]{5,}$/,
    USERNAME_INDIA: /^[a-zA-Z][a-zA-Z0-9]{5,}$/,
    USERNAME_SAUDIA: /^[a-zA-Z0-9]{8,}$/,
    USERNAME_EGYPT: /^[a-zA-Z]{10,}$/,
    PASSWORD: /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
    ALPHABETS_ONLY_2CHARS: /^[a-zA-Z]{2,}$/

};