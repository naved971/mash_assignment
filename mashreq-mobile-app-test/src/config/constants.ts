export const USERNAME_KEY = 'username';


export enum Languages {
    en = 'en',
    ja = 'ja',
    nl = 'nl'
}
export enum GlobalErrors {
    NetworkError = "Network Error"
}

export enum APIStatus {
    Error = "error",
    Success = "success"
}

export enum Countries {
    UAE = "UAE",
    India = "India",
    Egypt = "Egypt",
    Saudia = "Saudia"
}

export const CountryImages: any = {
    [Countries.UAE]: require('../assets/UAE.png'),
    [Countries.India]: require('../assets/India.png'),
    [Countries.Egypt]: require('../assets/Egypt.png'),
    [Countries.Saudia]: require('../assets/Saudia.png'),
}

export const LanguagesImages: any = {
    [Languages.en]: require('../assets/en.png'),
    [Languages.ja]: require('../assets/ja.png'),
    [Languages.nl]: require('../assets/nl.png'),
}