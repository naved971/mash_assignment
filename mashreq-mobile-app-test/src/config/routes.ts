export const ROUTE_USER_REGISTRATION = 'user_registration';
export const ROUTE_USER_DASHBOARD = 'user_dashboard';
