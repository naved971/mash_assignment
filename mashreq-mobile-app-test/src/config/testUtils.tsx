import React, { PropsWithChildren } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux'
import { setupStore } from '../redux/store'
import { render } from '@testing-library/react-native';


export const mockTheme = {
  // Your theme object goes here
  TextPrime: '#3d3d3d',
  ErrorTextAlert: '#cc0000',
  LightTextPrime: '#797979',
  SubtleHuePrimary: '#FFF3E0',
  LuminousPrimary: '#FFB74D',
  InactiveShade: '#D3D3D3',
  CoreChroma: "#FF9800",
  TextSecondary: "#ffffff"
};


export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: any = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
          <NavigationContainer>
            <ThemeProvider theme={mockTheme}>
              {children}
            </ThemeProvider>
          </NavigationContainer>
      </Provider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}



