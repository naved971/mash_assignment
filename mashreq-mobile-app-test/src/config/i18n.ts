/* eslint-disable */

import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import english from '../localization/en.json'
import dutch from '../localization/nl.json'
import japanese from '../localization/ja.json'
import { Languages } from './constants'

export const resources = {
  [Languages.en]: {
    translation: {
      ...english,
    }
  },
  [Languages.ja]: {
    translation: {
      ...japanese,
    }
  },
  [Languages.nl]: {
    translation: {
      ...dutch
    }
  }
}

i18n.use(initReactI18next).init({
  compatibilityJSON: 'v3',
  resources,
  lng: 'en',
  fallbackLng: 'en',
  interpolation: {
    escapeValue: false
  }
})

export default i18n
