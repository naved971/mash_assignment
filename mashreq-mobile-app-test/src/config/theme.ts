import { Countries } from "./constants";

export const base = {
    BackgroundCore: "#ffffff",
    LightBackgroundCore: "#f7f7f7",
    SecondaryBackdrop: "#3d3d3d",
    LightSecondaryBackdrop: "#797979",
    TextPrime: "#3d3d3d",
    LightTextPrime: "#797979",
    TextSecondary: "#ffffff",
    TextBackgroundPrime: "#ffffff",
    TextBackgroundSecondary: "#3d3d3d",
    InactiveShade: '#D3D3D3',
    ErrorTextAlert: '#cc0000'
};

const defaultPalettes =  {
    SubtleHuePrimary: "#E0F2F1",
    LuminousPrimary: "#4DB6AC",
    CoreChroma: "#009688",
    VibrantCore: "#00695C",
    ForegroundPrime: "#FFFFFF"
}

export const themePalettes = {
    [Countries.UAE]: {
        SubtleHuePrimary: "#FFDAB9",
        LuminousPrimary: "#FFA07A",
        CoreChroma: "#FF4500",
        VibrantCore: "#FF8C00",
        ForegroundPrime: "#FFFFFF"
    },
    [Countries.India]: {
        SubtleHuePrimary: "#FFE5B4",
        LuminousPrimary: "#FFFFFF",
        CoreChroma: "#138808",
        VibrantCore: "#FF4500",
        ForegroundPrime: "#000000"
    },
    [Countries.Egypt]: {
        SubtleHuePrimary: "#E6E6E6",
        LuminousPrimary: "#FFFFFF",
        CoreChroma: "#C8102E",
        VibrantCore: "#000000",
        ForegroundPrime: "#C8102E"
    },
    [Countries.Saudia]: {
        SubtleHuePrimary: "#DFF0D8",
        LuminousPrimary: "#FFFFFF",
        CoreChroma: "#008033",
        VibrantCore: "#154734",
        ForegroundPrime: "#000000"
    },
    default:defaultPalettes
};