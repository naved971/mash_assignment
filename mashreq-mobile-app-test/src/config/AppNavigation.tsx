import React from 'react';
import { NavigationContainer, DefaultTheme } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import RegisterUser from '../features/registerUser';
import Dashboard from '../features/dashboard';
import * as Routes from './routes';
import { ThemeProvider } from 'styled-components/native';
import { SafeAreaView, StyleSheet } from 'react-native';
import { useSelector } from 'react-redux';

const Stack = createNativeStackNavigator();

const AppNavigation = () => {
  const { theme } = useSelector((state: any) => state.themeReducer);
  const styles = useStyles(theme);

  const appTheme = {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      background: theme.BackgroundCore,
    },
  };

  return (
    <ThemeProvider theme={theme}>
      <SafeAreaView style={styles.safeAreaContainer}>
        <NavigationContainer theme={appTheme}>
          <Stack.Navigator
            screenOptions={{
              headerShown: false,
            }}
          >
            <Stack.Screen name={Routes.ROUTE_USER_REGISTRATION} component={RegisterUser} />
            <Stack.Screen name={Routes.ROUTE_USER_DASHBOARD} component={Dashboard} />
          </Stack.Navigator>
        </NavigationContainer>
      </SafeAreaView>
    </ThemeProvider>
  );
};

const useStyles = (theme: any) =>
  StyleSheet.create({
    safeAreaContainer: {
      flex: 1,
      backgroundColor: theme.BackgroundCore,
    },
  });

export default AppNavigation;
