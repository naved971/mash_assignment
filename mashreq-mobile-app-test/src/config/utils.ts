
import i18n from 'i18next'
import { I18nManager } from 'react-native';
import RNRestart from 'react-native-restart';
import { base, themePalettes } from "../config/theme";
import { Countries, GlobalErrors } from './constants';
import { AxiosError, AxiosResponse } from 'axios';
import SecureStore, { ACCESSIBLE } from "react-native-secure-key-store";

export const setLanguage = async (languageKey: string) => {
  const isNewLanguageRTL = i18n.dir(languageKey) === 'rtl';

  const isCurrentLayoutRTL = I18nManager.isRTL;

  const isLayoutChangeNeeded = isCurrentLayoutRTL !== isNewLanguageRTL;

  i18n.changeLanguage(languageKey)
  if (isLayoutChangeNeeded) {
    I18nManager.forceRTL(isNewLanguageRTL); // successful, since otherwise, the new
    RNRestart.restart();                    // language would not apply after restart.
  }
};

export const getTheme = (country?: string) => {
  switch (country) {
    case Countries.UAE:
      return { ...base, ...themePalettes[Countries.UAE] }
    case Countries.India:
      return { ...base, ...themePalettes[Countries.India] }
    case Countries.Egypt:
      return { ...base, ...themePalettes[Countries.Egypt] }
    case Countries.Saudia:
      return { ...base, ...themePalettes[Countries.Saudia] }
    default:
      return { ...base, ...themePalettes.default };
  }
}

export const handleNetworkError = (error: AxiosError) => {
  const axiosError = error as AxiosError;
  const response: AxiosResponse<any> | undefined = axiosError.response;
  const errorMessage = response?.data?.message || GlobalErrors.NetworkError;
  return errorMessage;
}


export const setKeyStore = async (key: string, value: string): Promise<void> => {
  try {
    await SecureStore.set(key, value, { accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY });
  } catch (error) {
    throw error;
  }
};

// Function to retrieve the username securely
export const getKeyStore = async (key: string): Promise<string | null> => {
  try {
    const value = await SecureStore.get(key);
    return value || null;
  } catch (error) {
    throw error;
  }
};

// Function to delete the saved username
export const deleteKeyStore = async (key: string): Promise<void> => {
  try {
    await SecureStore.remove(key);
  } catch (error) {
    throw error;
  }
};

