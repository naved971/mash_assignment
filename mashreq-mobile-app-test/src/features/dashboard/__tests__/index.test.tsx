import React from 'react';
import Dashboard from '../index';
import { renderWithProviders, mockTheme } from '../../../config/testUtils';
import { fireEvent, waitFor } from '@testing-library/react-native';
import * as Utils from '../../../config/utils';
import { useNavigation } from '@react-navigation/native'

const mockState = {
    userRegistration: {
        response: {
            status: "success",
            data: {
                country: "Inda",
                firstName: "Naved",
                lastName: "Khan"
            }
        },
        loading: false,
        error: null,
    }
};

jest.mock('styled-components', () => ({
    ...jest.requireActual('styled-components'),
    useTheme: () => (mockTheme),
}));


jest.mock('react-i18next', () => ({
    // mock useTranslation hook 
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: jest.fn(),
                language: 'EN'
            }
        }
    }
}))

jest.mock('react-redux', () => {
    return {
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn().mockImplementation((selectorFn) => selectorFn(mockState)),
        useDispatch: () => jest.fn(),
    };
});

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: ''
}))

// Mock the functions from '../../config/utils'
jest.mock('../../../config/utils', () => ({
    getKeyStore: jest.fn(),
    deleteKeyStore: jest.fn(),
    getTheme: jest.fn(),  // Mock the getTheme function
}));

jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: jest.fn(),
}));

const mockedUtils = Utils as jest.Mocked<typeof Utils>;


test('test dashboard', async () => {


    const mockNavigation = {
        navigate: jest.fn(),
        dispatch: jest.fn(),
    };
    (useNavigation as jest.Mock).mockReturnValue(mockNavigation);
    
    const { toJSON, getByTestId } = renderWithProviders(<Dashboard />);

    mockedUtils.getKeyStore.mockResolvedValue('mockUsername');

    await waitFor(() => {
        expect(mockedUtils.getKeyStore).toHaveBeenCalledWith('username');
    });

    mockedUtils.deleteKeyStore.mockResolvedValue();


    fireEvent.press(getByTestId('logOut'));

    await waitFor(() => {
        expect(mockedUtils.deleteKeyStore).toHaveBeenCalledWith('username');
    });

    expect(toJSON()).toMatchSnapshot();
});