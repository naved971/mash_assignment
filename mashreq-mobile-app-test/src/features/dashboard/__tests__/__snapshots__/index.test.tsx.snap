// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dashboard 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 22,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <
        color="#FF9800"
        icon={
          {
            "icon": [
              512,
              512,
              [
                62142,
                "user-circle",
              ],
              "f2bd",
              "M399 384.2C376.9 345.8 335.4 320 288 320H224c-47.4 0-88.9 25.8-111 64.2c35.2 39.2 86.2 63.8 143 63.8s107.8-24.7 143-63.8zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256 16a72 72 0 1 0 0-144 72 72 0 1 0 0 144z",
            ],
            "iconName": "circle-user",
            "prefix": "fas",
          }
        }
        size={72}
      />
      <View
        style={
          {
            "marginLeft": 6,
          }
        }
      >
        <Text
          color="#3d3d3d"
          style={
            [
              {
                "color": "#3d3d3d",
              },
              {
                "fontSize": 18,
                "fontWeight": "600",
              },
            ]
          }
        >
          welcome null!
        </Text>
        <View
          style={
            {
              "marginTop": 8,
            }
          }
        >
          <Text
            color="#3d3d3d"
            style={
              [
                {
                  "color": "#3d3d3d",
                },
                {
                  "fontSize": 14,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Naved Khan
          </Text>
          <Text
            color="#FF9800"
            style={
              [
                {
                  "color": "#FF9800",
                },
                {
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Saudia
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FF9800",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "height": 54,
          "justifyContent": "center",
          "marginVertical": 8,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        }
      }
      testID="logOut"
    >
      <Text
        color="#ffffff"
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        logOut
      </Text>
    </View>
  </View>
</View>
`;
